iree/runtime/demo/hello_world_terse.c

```diff
@@ -83,8 +83,12 @@ static void iree_runtime_demo_perform_mul(iree_runtime_session_t* session) {
       iree_runtime_session_device_allocator(session), arg0_shape,
       IREE_ARRAYSIZE(arg0_shape), IREE_HAL_ELEMENT_TYPE_FLOAT_32,
       IREE_HAL_ENCODING_TYPE_DENSE_ROW_MAJOR,
-      IREE_HAL_MEMORY_TYPE_HOST_LOCAL | IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE,
-      IREE_HAL_MEMORY_ACCESS_READ, IREE_HAL_BUFFER_USAGE_ALL,
+      (iree_hal_buffer_params_t){
+          .type = IREE_HAL_MEMORY_TYPE_HOST_LOCAL |
+                  IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE,
+          .access = IREE_HAL_MEMORY_ACCESS_READ,
+          .usage = IREE_HAL_BUFFER_USAGE_ALL,
+      },
```

Comment:
I was a little worried about these functions getting even more verbose, but this syntax with named initializers for the struct reads pretty easily.

Reply:
intellisense works pretty well too, and now (with the coming changes) that the fields are mostly optional these could even just be `..., {IREE_HAL_BUFFER_USAGE_FOO}, ...` inline. C++ 20 will be able to do this named stuff too, but for now that unnamed style should at least make that cleaner 🤞